pkg/controller/ehpa/hpa.go

```diff
@@ -185,34 +185,16 @@ func (c *EffectiveHPAController) GetHPAMetrics(ctx context.Context, ehpa *autosc
 		var metricsForPrediction []autoscalingv2.MetricSpec
 
 		for _, metric := range metrics {
-			// generate a custom metric for resource metric
-			if metric.Type == autoscalingv2.ResourceMetricSourceType {
-				name := utils.GetPredictionMetricName(metric.Resource.Name)
-				if len(name) == 0 {
-					continue
+			var metricIdentifier string
+			var averageValue *resource.Quantity
+			switch metric.Type {
+			case autoscalingv2.ResourceMetricSourceType:
+				switch metric.Resource.Name {
```

Comment:
metricIdentifier = utils.GetMetricIdentifier(metric, metric.Resource.name.String())

no need for switch case

Reply:
done